### PR TITLE
Test on Ruby 2.4 and fix failing tests

### PR DESCRIPTION
**Pull request by tas50**
Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -2,7 +2,7 @@ require "foodcritic/version"
 require "bundler"
 require "rake/testtask"
 
-task :default => [:man, :install, :rubocop, :test, :features]
+task :default => [:man, :install, :test, :features, :rubocop]
```
Comment (josephholsten) [inline Rakefile]: usually I sort these by how long it takes to run, so I'm a bit confused by this change

Comment (tas50) [inline Rakefile]: This is something lamont brought up today and I was running into it. Style is the thing you care about the least and you often fail tests there first, which stops testing. This way you see your failing specs first and you can clean up your style later. It avoids multiple commits to style.

**Comment by tas50** [comment]
I've removed the tests that check for the exist status as these all fail now. Our default behavior is now to fail on any failing tag. If we generate a cookbook that lacks basic things like metadata it will ALWAYS fail a tag and thus always exit non-0. There's no way to test this in the current design of the tests / app.

File: Rakefile (hunk 1 above)
Comment (lamont-granquist) [inline Rakefile]: yeah it annoys the crap out of me to have rubocop fail first because fixing that is usually mindless.

**Pull request merged**